api/envoy/extensions/filters/http/ext_proc/v3/processing_mode.proto

```diff
@@ -68,6 +69,31 @@ message ProcessingMode {
     // chunk. If the body exceeds the configured buffer limit, then the body contents
     // up to the buffer limit will be sent.
     BUFFERED_PARTIAL = 3;
+
+    // [#not-implemented-hide:]
+    // Envoy streams the body to the server in pieces as they arrive. The data is not stored in
+    // the internal queue. Only data size is stored there for tracking and flow control purpose.
+    //
+    // Server may choose to buffer any number chunks of data before processing them.
+    // After finish buffering, server processes the buffered data. Then it splits the processed
+    // data into any number of chunks, and stream them back to Envoy one by one.
+    // The server may continuously do so until the complete body is processed.
+    //
+    // Server may also choose to buffer the entire message, including the headers(if header mode is
+    // ``SEND``), the entire body, and the trailers(if present), before sending back any response.
+    // The server response has to maintain the headers-body-trailers ordering.
+    //
+    // In this body mode:
+    // * The corresponding trailer mode has to be set to ``SEND``.
+    // * Envoy will send body and trailers(if present) as they arrive.
+    //   Sending the trailers(if present) is to inform the server the complete body arrives.
+    //   In case of no trailers, then
+    //   :ref:`end_of_stream <envoy_v3_api_field_service.ext_proc.v3.HttpBody.end_of_stream>`
+    //   serves for that purpose.
+    // * Server needs to send
+    //   :ref:`StreamedBodyResponse <envoy_v3_api_msg_service.ext_proc.v3.StreamedBodyResponse>`
+    //   to Envoy in the body response.
+    BIDIRECTIONAL_STREAMED = 4;
```

Comment:
Overall, the above seems to be describing the good-path.
What happens when one of the actors isn't playing well (e.g., the server sends end-stream before receiving the entire body or trailers).

Reply:
If the server is behaving like that, the garbage request will be send to backend server. That's not something Envoy can do and verify.